contributors/devel/api_changes.md

```diff
@@ -927,8 +927,10 @@ The preferred approach adds an alpha field to the existing object, and ensures i
     }
     ```
 
-3. Before persisting the object to storage, clear disabled alpha fields.
-One possible place to do this is in the REST storage strategy's PrepareForCreate/PrepareForUpdate methods:
+3. Before persisting the object to storage, clear disabled alpha fields on create,
+and on update if the existing object does not already have a value in the field.
```

Comment:
Is the intention to return gated-off fields on GETs?  Why?  Shouldn't they be filtered out if the gate is off?

Reply:
> Is the intention to return gated-off fields on GETs? Why?

Once data is validated/persisted, the safest thing we can do is round trip and return it intact, and let objects with data from a disabled feature be deleted via normal processes. See earlier comments on dropping data in gated fields from existing persisted objects not being uniformly safe.

As an aside, we already return persisted data as-is on GET now, regardless of feature gate. The changes in this PR ensure a persisted object will not become invalid or impossible to delete as a result of rollback or feature disablement.